### PR TITLE
Fix compile warnings/errors with clang14 and gcc11

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

- [X] :hammer: Refactoring/simplification

### CODE REVIEW

File: arangod/IResearch/AqlHelper.cpp [hunk 1]
```diff
@@ -646,7 +646,7 @@ bool attributeAccessEqual(aql::AstNode const* lhs, aql::AstNode const* rhs,
 
   // TODO: is the "&" intionally. If yes: why?
   // cppcheck-suppress uninitvar; false positive
-  while (lhsValue.read(lhs, ctx) & rhsValue.read(rhs, ctx)) {
+  while (lhsValue.read(lhs, ctx) & (int)rhsValue.read(rhs, ctx)) {
```
Comment (MBkkt) [inline arangod/IResearch/AqlHelper.cpp:649]: I don't like fix:
c style cast
signed integer operand with a binary bitwise operator

**Review by MBkkt — APPROVED** [review]

File: arangod/IResearch/AqlHelper.cpp (hunk 1 above)
Comment (goedderz) [reply arangod/IResearch/AqlHelper.cpp:649]: Good points. Would
```suggestion
  while (lhsValue.read(lhs, ctx) &
         static_cast<unsigned>(rhsValue.read(rhs, ctx))) {
```
work for you? Alternatively, we could go more verbose like
```cpp
  bool lhsRead = lhsValue.read(lhs, ctx);
  bool rhsRead = rhsValue.read(rhs, ctx);
  while (lhsRead && rhsRead) {
    if (lhsValue != rhsValue) {
      return false;
    }

    lhs = lhsValue.root;
    rhs = rhsValue.root;
    lhsRead = lhsValue.read(lhs, ctx);
    rhsRead = rhsValue.read(rhs, ctx);
  }
  ```

Comment (goedderz) [reply arangod/IResearch/AqlHelper.cpp]: Alright thanks, I've applied the first suggestion in https://github.com/arangodb/arangodb/pull/16400/commits/7e0ed87cb12c58e3299d1a50adb9f0d11527eeb1.

File: arangod/IResearch/AqlHelper.cpp [hunk 2]
```diff
@@ -646,7 +646,8 @@ bool attributeAccessEqual(aql::AstNode const* lhs, aql::AstNode const* rhs,
 
   // TODO: is the "&" intionally. If yes: why?
   // cppcheck-suppress uninitvar; false positive
-  while (lhsValue.read(lhs, ctx) & rhsValue.read(rhs, ctx)) {
+  while (lhsValue.read(lhs, ctx) &
```
Comment (maierlars) [inline arangod/IResearch/AqlHelper.cpp:649]: now I wonder as well 🤔

**Review by maierlars — APPROVED** [review]

**Pull request merged**